scripts/README.md

```diff
@@ -1,56 +1,85 @@
 # FCREPO3 Analysis Helpers
-
 ## Introduction
-Scripts to analyse and export metadata from an FCREPO3 instance.
+Tools to analyse and export metadata from an FCREPO3 instance using Python scripts.
 
 ## Table of Contents
-
 * [Setup](#setup)
 * [Features](#features)
 * [Usage](#usage)
 
 ## Setup
+These tools are designed to be run with a Python environment. Ensure Python 3.6 or higher is installed on your system. You will need to set up a Python virtual environment and install the required packages:
 
-These scripts require an FCREPO3 instance to be run over. In the event, these scripts are run on a separate system from where
-the repository lives, modifications may be required to the `fedora-xacml-policies` directory located at `$FEDORA_HOME/data/fedora-xacml-policies`.
+```bash
+python3 -m venv venv
+source venv/bin/activate
+pip install -r requirements.txt
+```
 
-The metadata export command requires [GNU Parallel](https://www.gnu.org/software/parallel/parallel.html) to be installed
-for faster processing.
+The scripts also require an FCREPO3 instance. If these tools are run on a system separate from where the repository is hosted, modifications might be necessary in the `fedora-xacml-policies` directory at `$FEDORA_HOME/data/fedora-xacml-policies`.
 
 ## Features
-
 ### Metadata Analysis
-A script to generate the following:
-1. A total count of all objects in the repository.
-2. A breakdown of objects by content models and their count in CSV form (`models.csv`).
-3. A breakdown of unique datastream IDs and their count in CSV form (`dsids.csv`).
+Script to run SPARQL queries against an FCREPO's RI and gather information. Current queries include:
+ - Content model distribution
+ - Total object count
+ - Count of active and deleted objects
+ - List of deleted objects
+ - Datastream distribution
+ - Owner distribution
+ - Collection distribution
+ - List of relationships
+ - List of orphaned objects
+ - MIME type distribution
 
 ### Metadata Export
-A script to export all objects within the repository that contain a specified metadata datastream ID.
+Script to export all objects within the repository that contain a specified metadata datastream ID, saving results as XML.
 
-## Usage
+### FOXML Export
+Script to export FOXML archival objects from a Fedora repository given a list of PIDs.
 
+### Datastream Updater
+Script to inject a binary into an archival FOXML as base64 encoded data within a datastream.
+
+## Usage
 ### Metadata Analysis
```

Comment:
Before running this as a script, you might want to verify that the queries provided in queries.py are compatible with the system you are querying. If the system has Mulgara instead of Blazegraph, it would be restricted to SPARQL 1.0; to check what features would not be available for SPARQL 1.0, you can see a list of new features added in SPARQL 1.1 at the bottom of [https://www.w3.org/TR/sparql11-query/](https://www.w3.org/TR/sparql11-query/).
If you find that the system you are querying has some relationships outside of the ones covered in these queries by default, you will need to modify the relevant queries to get an accurate analysis. For example, sometimes there are relationships that have different capitalization or typos; these relationships need to be accounted for in this analysis phase for completeness and accuracy of the analysis as well as ensuring they get mapped appropriately in the actual migration.